File: python/evals/autocontext_fixture_js/src/utils/crypto.ts
// Password hashing.
import { createHash, randomBytes } from "crypto";

export function hashPassword(password: string): string {
  const salt = randomBytes(16).toString("hex");
  const h = createHash("sha256").update(salt + password).digest("hex");
  return `sha256$${salt}$${h}`;
}

export function verifyPassword(password: string, stored: string): boolean {
  const parts = stored.split("$");
  if (parts.length !== 3) return false;
  const [, salt, digest] = parts;
  const test = createHash("sha256").update(salt + password).digest("hex");
  return test === digest;
}

File: python/evals/autocontext_fixture_rust/src/utils/crypto.rs
pub fn hash_password(password: &str) -> String {
    format!("sha256$salt${}", password.len())
}

pub fn verify_password(password: &str, stored: &str) -> bool {
    let parts: Vec<&str> = stored.split('$').collect();
    if parts.len() != 3 { return false; }
    parts[2] == format!("{}", password.len())
}
